src/load.ts

```diff
@@ -0,0 +1,62 @@
+import { utils } from "@pinecone-database/pinecone";
+import cliProgress from "cli-progress";
+import { config } from "dotenv";
+import loadCSVFile from "./csvLoader.js";
+
+import { embedder } from "./embeddings.js";
+import { getPineconeClient } from "./pinecone.js";
+import { getEnv } from "./utils/util.js";
+const { createIndexIfNotExists, chunkedUpsert } = utils;
+config();
+
+const progressBar = new cliProgress.SingleBar(
+  {},
+  cliProgress.Presets.shades_classic
+);
+
+let counter = 0;
+
+export const load = async (csvPath: string, column: string) => {
+  // Get index name
+  const indexName = getEnv("PINECONE_INDEX");
+
+  // Get a PineconeClient instance
+  const pineconeClient = await getPineconeClient();
+
+  // Create a readable stream from the CSV file
+  const { data, meta } = await loadCSVFile(csvPath);
+
+  // Ensure the selected column exists in the CSV file
+  if (!meta.fields?.includes(column)) {
+    console.error(`Column ${column} not found in CSV file`);
+    process.exit(1);
+  }
+
+  // Extract the selected column from the CSV file
+  const documents = data.map((row) => row[column] as string);
+
+  // Create a Pinecone index with the name "word-embeddings" and a dimension of 384
+  await createIndexIfNotExists(pineconeClient, indexName, 384).then(
+    // Give a little more space for index to be up and running
+    () => new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
This should assume the index has been created correctly, and remove the additional `setTimeout`

Reply:
Sure, I removed timeouts. Please test on your machine. On mine, it is working 4/5 times.